### DISCUSSION TITLE
Sort columns by stat in histplot

### DISCUSSION THREAD

**Discussion by baggiponte** [mwaskom/seaborn#3311] [Q&A] [answered]
I am using `sns.histplot` and I would like to sort the columns in descending order by size - I googled around and could not find much on stackoverflow and skimming through issues in the repo.

![image](https://user-images.githubusercontent.com/57922983/229756309-225ffcbf-db05-457b-86ee-dab2991f4559.png)

Could this be a feature you'd be interested in having in seaborn?

**Comment by Poslovitch** [comment]
would appreciate this.

**Comment by mwaskom** [comment, marked as answer]
It makes sense but it would be difficult-to-impossible to implement given the way the categorical unit handling works (the mapping between strings and integer positions is obtained before computing the statistics).

If you compute the desired ordering yourself and use that to assign a Categorical dytpe, it should be respected.
